src/service.js

```diff
@@ -0,0 +1,73 @@
+import request from 'request';
+import promisify from './promisify';
+import ConsultantError from './consultant-error';
+import properties from './properties';
+import fetchIdentifier from './identifier';
+
+const deregister = async (host, instance) => {
+	const req = {
+		url : `${host}/v1/agent/service/deregister/${instance.instance}`,
+		method : 'DELETE',
+		headers : {
+			'user-agent' : properties.userAgent
+		}
+	};
+
+	try {
+		await promisify(cb => request(req, cb));
+	}
+	catch (e) {
+		throw new ConsultantError(`Could not deregister from Consul: ${e}`);
+	}
+};
+
+export default async function register({service, healthCheckPath, healthCheckInterval = 10, consulHost}) {
+	consulHost = consulHost || process.env.CONSUL_HOST || properties.defaultHost;
+
+	if (!Number(service.port)) {
+		throw new ConsultantError('service.port was not defined properly');
```

Comment:
ok now I'm getting annoying, but also check whether the port matches `0 < port < 65536`